packages/expo-sms/ios/ExpoSMSModule.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  ExpoSMSModule.swift
+//  ExpoSMS
+//
+//  Created by Alan Hughes on 11/11/2022.
+//
+
+import Foundation
+import ExpoModulesCore
+import WebKit
+import MessageUI
+import CoreServices
+import MobileCoreServices
+import UniformTypeIdentifiers
+
+struct ExpoSMSContext {
+    let promise: Promise
+    let smsDelgate: SMSDelegate
+}
+
+public class ExpoSMSModule: Module, SMSResultHandler {
+    private var smsContext: ExpoSMSContext?
+    private lazy var utils = appContext?.utilities
+
+    public func definition() -> ModuleDefinition {
+        Name("ExpoSMS")
+
+        AsyncFunction("isAvailableAsync") { (promise: Promise) in
+            promise.resolve(MFMessageComposeViewController.canSendText())
+        }
+
+        AsyncFunction("sendSMSAsync") {(addresses: [String], message: String, options: SmsOptions?, promise: Promise) in
+            sendSMSAsync(addresses: addresses, message: message, options: options, promise: promise)
+        }.runOnQueue(.main)
+    }
+
+    private func sendSMSAsync(addresses: [String], message: String, options: SmsOptions?, promise: Promise) {
+        if !MFMessageComposeViewController.canSendText() {
+            promise.reject("E_SMS_UNAVAILABLE", "SMS service not available")
+            return
```

Comment:
Similarly to `isAvailableAsync` where we can just return the value without a promise, here we can just throw an exception.
```suggestion
    private func sendSMSAsync(addresses: [String], message: String, options: SmsOptions?, promise: Promise) throws {
        if !MFMessageComposeViewController.canSendText() {
            throw SmsUnavailableException()
```

And add the implementation for `SmsUnavailableException` somewhere:
```swift
internal class SmsUnavailableException: Exception {
  override var reason: String {
    "SMS service is not available"
  }
}
```